frontend/index.html

```diff
@@ -722,53 +787,57 @@ <h2 data-i18n="contact">Contact</h2>
+        
+        //load the dynamic config
         const dynamicLoadPromise = fetch(CONFIG_URL).then((resp) => resp.json());
+        console.log("Dynamic Config Promise", dynamicLoadPromise);
+        data = await dynamicLoadPromise; 
+        //don't continue until data is defined and config is loaded
+        console.log("Dynamic Config result is", data);
+
+        //create token and token url
+        OPCODE_AUTOGEN = data.opcode?.autogen || data.intro.program_or_study == "study";
+        if (OPCODE_AUTOGEN) {
+          validateAndGenerateToken();
+        } else {
+          hideTokenAndURL();
+        }
+        SUPPORTED_LANGUAGES = getSupportedLanguages();
+
+        //load the default (first listed in translated) language
+        const default_language = SUPPORTED_LANGUAGES[0];//default should be set up to be the first language!
+        console.log("language default is", default_language);
+        curr_lang = default_language;
+          
         // i18n.load returns a jquery promise, but apparently, since jquery3,
         // they are ES2015 compatible, so we don't need to wrap them
         // also we first load only the default language, so we can display the page ASAP
-        const i18nLoadPromise = $.i18n().load('i18n/'+curr_lang+'.json', curr_lang);
-        const allAsyncLoadPromises = [i18nLoadPromise, dynamicLoadPromise];
-        console.log("First round promises", allAsyncLoadPromises);
-
-        Promise.all(allAsyncLoadPromises).then(([i18nLoadResult, dynamicLoadResult]) => {
-            console.log("Async results are ", i18nLoadResult, dynamicLoadResult);
-            data = dynamicLoadResult;
-            // TODO: Remove backwards compatibility at the end of 2024, once all
-            // current programs have completed
-            OPCODE_AUTOGEN = data.opcode?.autogen || data.intro.program_or_study == "study";
-            if (OPCODE_AUTOGEN) {
-                validateAndGenerateToken();
-            } else {
-                hideTokenAndURL();
-            }
-            redisplayPage();
-        }).catch((error) => setInvalid($.i18n("errorLoadingPageElements"),
-            error, $.i18n("contactForAssistance", "k.shankari@nrel.gov")));
-        // the error elements are hardcoded here because we may have an error
-        // in loading the i18n, in which case we will not have i18n-ized text
-        // available to display.
-        // END: First round of loading
-
-        // BEGIN: Second round of loading
-        // then, we load all the other languages
+        const defaultI18nLoadPromise = $.i18n().load('i18n/'+default_language+'.json', default_language);
+        console.log("Default i18n Promise", defaultI18nLoadPromise); //supposed to be undefined? works and looks undefined in the original code?
+        const i18nLoadResult = await defaultI18nLoadPromise;
+        console.log("Default i18n result is", i18nLoadResult);
+
+        $('#langSel').val(default_language);
```

Comment:
For my own edification, `#langSel` won't actually have any entries here, right? (that will happen in `makeLanguageMenu` after the other promises are loaded).

So I don't think this will actually do anything. It doesn't hurt anything, so I am OK to merge now, but you might want to test that out to understand the code better and polish it further.

Unless you think that the `val` is stored somehow and applied when the values are added in which case, document that for the next person working on this code.

Reply:
You were absolutely right, I moved this line of code until after the language is created and enabled (after the other promises are loaded). This lets the name of the language still be displayed in the menu as soon as the menu is ready! The change is reflected in #19, which can address these and any other polishing changes we decide to make.